### DISCUSSION TITLE
Uploading non-image assets

### DISCUSSION THREAD

**Discussion by aidabeorn** [requarks/wiki#3895] [Error / Bug Report] [answered]
When uploading a non-image asset, such as a text document (.txt) the link to the document returns a 404.

This results in confusing behavior for end-users, as there's little indication that an asset has to be an image or video.
For the way we use the wiki, non-media assets are desirable to have.

steps to repro:
1. Create a new text document, content doesn't really matter
2. Upload the document to a wiki.js instance
3. Insert the asset to your page
4. Save
5. Follow the link

**Comment by StormyIceLeopard** [comment]
I am having the same issue.

**Comment by billytetrud** [comment]
[The documentation](https://docs.requarks.io/guide/assets) says "The assets can be of any file type; images, documents, zip, etc." I'm seeing the same problem.

**Comment by NGPixel** [comment, marked as answer]
Update to 2.5.285 and remove `txt` from the Page Extensions field under Administration > General

**Comment by yoitguy** [comment]
I tested the txt work-around mentioned by NGPixel and it works well. But this doesn't help with PDFs. I can upload images and txt files now, and everything works well. But after uploading a pdf and putting a link into a page, the wiki will return "It looks like the webpage at https://itwikidev.myfakedomain23232.com/hsmguide.pdf might be having issues, or it may have moved permanently to a new web address." If this should be working, how can I troubleshoot?

**Reply by NGPixel** [reply]
You can try turning off the "Force Download of Unsafe Extensions" option under Admin > Security.

**Reply by yoitguy** [reply]
Thank you for the suggestion. I tried turning off this option and it still fails, but the error has changed to "Internal Server Error." As I test, I temporarily disabled the rest of the security features as well, to no avail. Are there any logs I can view to troubleshoot?

**Reply by yoitguy** [reply]
Does Wiki.js rely on a webserver to serve up attachments such as this, for example apache or is it 100% self-contained to node.js as is my understanding? It acts as though the attachment isn't there.

**Reply by NGPixel** [reply]
There's no webserver in Wiki.js. You're interacting directly with node.js HTTP server.

Check the logs from the process output (or container output if using docker).

**Reply by yoitguy** [reply]
Not having any node.js experience, I was able to do a web search and find that I can enable logging by starting wiki.js with "node --inspect server" which is outputting the logs to stdout. From here, I tried to access the PDF and noticed the following error is generated: "[MASTER] error: Cannot read properties of undefined (reading 'data')

It's the same error whether the "Force Download of Unsafe Extensions" is turned on or off.

This may be a silly question, but can you please confirm that PDFs work for you? Also, anything else that I can try?

**Reply by yoitguy** [reply]
OK, after various rounds of troubleshooting, I may have found a bug. I found that small PDFs do work. I experimented with various sizes and have seemed to narrow it down to breaking somewhere between 512kB pdfs (works) and 518kB PDFs (does not work). The methodology that I used was that I created a document within MS Word and printed it to a PDF. I varied the sizes of my files by increasing or decreasing the sizes of images within the document before printing it to a PDF.

**Reply by yoitguy** [reply]
Final update: After migrating from MariaDB to Postgres, the problem is resolved completely.
